composer/datasets/cifar.py

```diff
@@ -49,10 +69,79 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
                 memory_format=self.synthetic_memory_format,
             )
 
+        elif self.use_ffcv:
+            try:
+                import ffcv  # type: ignore
+                from ffcv.fields.decoders import IntDecoder, SimpleRGBImageDecoder  # type: ignore
+                from ffcv.pipeline.operation import Operation  # type: ignore
+            except ImportError:
+                raise ImportError(
+                    textwrap.dedent("""\
+                    Composer was installed without ffcv support.
+                    To use ffcv with Composer, please install ffcv in your environment."""))
+
+            dataset_file = self.ffcv_dest_train if self.is_train else self.ffcv_dest_val
+            dataset_filepath = os.path.join(self.ffcv_dir, dataset_file)
+            # always create if ffcv_write_dataset is true
+            if dist.get_global_rank() == 0 and self.ffcv_write_dataset:
+                if self.datadir is None:
+                    raise ValueError("datadir is required if use_synthetic is False and ffcv_write_dataset is True.")
+                ds = CIFAR10(
+                    self.datadir,
+                    train=self.is_train,
+                    download=self.download,
+                )
+
+                create_ffcv_dataset(dataset=ds, write_path=dataset_filepath)
+
```

Comment:
A `dist.barrier()` is needed here to ensure that rank zero has finished downloading before rank one attempts to read it.